tests/unit/api/test_feature.py

```diff
@@ -799,6 +807,76 @@ def test_feature__as_default_version(saved_feature):
     assert saved_feature.is_default is False
 
 
+def check_offline_store_ingest_graph_on_composite_feature(feature_model):
+    """Check offline store ingest graph on composite feature"""
+    # case 1: no entity relationship
+    assert feature_model.relationships_info == []
+    ingest_query_graphs = feature_model.extract_offline_store_ingest_query_graphs()
+
+    # check the first offline store ingest query graph
+    assert len(ingest_query_graphs) == 2
+    if ingest_query_graphs[0].node_name == "project_1":
+        ingest_query_graph1 = ingest_query_graphs[0]
+        ingest_query_graph2 = ingest_query_graphs[1]
+    else:
+        ingest_query_graph1 = ingest_query_graphs[1]
+        ingest_query_graph2 = ingest_query_graphs[0]
+
+    assert ingest_query_graph1.graph.edges_map == {
+        "input_1": ["graph_1"],
+        "graph_1": ["groupby_1"],
+        "groupby_1": ["project_1"],
+    }
+    out_node = ingest_query_graph1.graph.get_node_by_name(ingest_query_graph1.node_name)
+    assert isinstance(out_node, ProjectNode)
+    assert out_node.parameters.columns == ["sum_30m_by_binary"]
+
+    # check the second offline store ingest query graph
+    assert ingest_query_graph2.node_name == "add_1"
```

Comment:
Currently the offline store features represented by the decompose graphs don't have names, is that right? Curious how will that be handled? Is that by inserting alias nodes into the decomposed graphs?

Reply:
Offline feature name will be handled in a separate PR. Curious whether it is possible to include a feature name at `OfflineStoreIngestQueryGraph` and let's SQL generation to handle that? I am thinking of adding a graph consistency check by doing the following asserting
```
flat_origin_graph = flatten(original_graph)
flat_decom_graph = flatten(decomposed_graph)
assert flat_origin_graph.node_name_to_ref[flat_origin_node_name] == flat_decom_graph[flat_decom_node_name]
```

If we insert an alias node at the offline feature ingest graph, it will make it inconsistency check a bit tedious.

Reply:
That should be possible. If there are alias nodes, we can avoid special handling in SQL generation since that is the current way of encoding feature names. But that can be done outside of feature ingest graph extraction, or we can add some special handling if there are no better ways.